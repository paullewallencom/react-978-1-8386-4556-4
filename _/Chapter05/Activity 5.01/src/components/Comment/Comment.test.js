import React from "react";
import { render, cleanup } from "@testing-library/react";
import { Comment } from "./index";

afterEach(cleanup);

const withMarkup = query => text =>
  query((content, node) => {
    const hasText = node => node.textContent === text;
    const childrenDontHaveText = Array.from(node.children).every(
      child => !hasText(child)
    );
    return hasText(node) && childrenDontHaveText;
  });

const comment = {
  name: "Karen Michaels",
  text:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean scelerisque, purus ac feugiat eleifend, ex.",
  image:
    "https://images.unsplash.com/photo-1567336273898-ebbf9eb3c3bf?auto=format&fit=crop&w=200&q=80",
  time: "Oct 10, 2019"
};

it("renders comment", () => {
  const { asFragment } = render(<Comment comment={comment} />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders name", () => {
  const { getByText } = render(<Comment comment={comment} />);
  expect(!!withMarkup(getByText)(comment.name)).toBe(true);
});

it("renders text", () => {
  const { getByText } = render(<Comment comment={comment} />);
  expect(!!withMarkup(getByText)(comment.text)).toBe(true);
});

it("renders time", () => {
  const { getByText } = render(<Comment comment={comment} />);
  expect(!!withMarkup(getByText)(comment.time)).toBe(true);
});
