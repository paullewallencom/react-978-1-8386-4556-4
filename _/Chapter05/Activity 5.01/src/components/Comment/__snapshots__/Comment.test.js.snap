// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders comment 1`] = `
<DocumentFragment>
  <div
    class="comment"
  >
    <div
      class="comment_container"
    >
      <img
        alt="Karen Michaels"
        class="comment_profile"
        src="https://images.unsplash.com/photo-1567336273898-ebbf9eb3c3bf?auto=format&fit=crop&w=200&q=80"
      />
      <div
        class="comment_content"
      >
        <h2
          class="comment_name"
        >
          Karen Michaels
        </h2>
        <strong
          class="comment_time"
        >
          Oct 10, 2019
        </strong>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean scelerisque, purus ac feugiat eleifend, ex.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
